### PR TITLE
rewrite lua queries to match the updated lua parser

### PR DESCRIPTION
**Pull request by kyazdani42**
lua highlighting now works perfectly with [this PR](https://github.com/nvim-treesitter/tree-sitter-lua/pull/2) and the updated version of the query file. I didn't take time to look at the other lua files, i'll do it tomorrow. 
Basically the fix for lua highlighting is to declared function and end as keywords because `function` is an identifier but also the keyword `"function"` that we had to match literally. This was messing up the parser badly.

### CODE REVIEW

File: queries/lua/highlights.scm [hunk 1]
```diff
@@ -7,8 +7,8 @@
 [
   "if" 
   "then"
-  "end"
```
Comment (stsewd) [inline queries/lua/highlights.scm:10]: you could add `(end_keyword)` here instead of writing another line. Same for the others

**Comment by vigoux** [comment]
I left some comments on the lua parser changes, I'll review that right after those comments are adressed.

**Comment by vigoux** [comment]
If the nodes are hidden, they can be referenced with the `_` prepended, and everything should work great as before.

**Comment by kyazdani42** [comment]
@vigoux you can review

**Comment by kyazdani42** [comment]
> If the nodes are hidden, they can't be referenced with the `_` prepended, and everything should work great as before.

not sure what you mean here

**Comment by vigoux** [comment]
I did some testing, and I really don't see why we can't just capture the `"end"` anymore...

**Comment by stsewd** [comment]
You can't use anonymous nodes in queries `(_node)`, but we can capture literals `end`, so looks like we don't need this PR.

**Comment by kyazdani42** [comment]
well it did not work at all previously, this is why i updated the parser and i made it work properly in every file, so yes, we do need this PR.

**Comment by vigoux** [comment]
For some reason, we can't match the `"end"` now, and I really don't know why.

**Comment by kyazdani42** [comment]
actually we do not need to update the parser indeed. The query rules were just wrong. I'm going to update them in another PR

**Pull request closed without merging**